src/hooks/stations.ts

```diff
@@ -48,10 +48,27 @@ export const useBooking = () => {
         }),
     [BUCHUNGEN],
   );
+  const refreshBooking = useCallback(
+    async () => {
+      const oldBooking = await getCurrentBooking();
+      return cancelBooking()
+        .then(() => {
+          if (oldBooking) {
+            doBook(oldBooking.stationId);
```

Comment:
`doBook` returns the booking, so you should just set the state using that instead of re-fetching below.

Reply:
See 09a60998e6e830d41f6d86597266735b569ac1e8.